packages/xdl/package.json

```diff
@@ -90,6 +90,7 @@
     "terminal-link": "^2.1.1",
     "text-table": "^0.2.0",
     "tree-kill": "1.2.2",
+    "ts-node": "^10.2.1",
```

Comment:
keep in mind the bundle size increase by including [`ts-node`](https://packagephobia.com/result?p=ts-node), this isn't _too_ big relative to expo-cli, but it also depends on the user having typescript globally installed on their computer (which we aren't checking for currently), or locally packaged in the CLI ([typescript size](https://packagephobia.com/result?p=typescript) (non-typescript users would be installing a lot of code that they don't need)). It's also unclear whether this only supports the main `webpack.config.ts` file (i.e. all imports must be transpiled some other way), or what the `tsconfig.json` resolution would be since ts-node would be executed from a higher context.

Not being able to customize `tsconfig.json` could be a problem for some users, especially when new language features come out. IIRC `ts-node` is also quite strict by default, and slow to transpile, which is why they've been exploring [swc support](https://github.com/TypeStrong/ts-node#bundled-swc-integration), have you measured the net time increase (just in general, nothing fancy)?

---

Seems like maybe most of this could be better off, by living in user-space for now, and reusing the existing `tsconfig.json` and `typescript` modules that a TypeScript user has installed in their project. 

They could then have a TypeScript Webpack config with:

`webpack.config.js`
```js
require('ts-node/register');
module.exports = require('./webpack.config.ts')
```

`webpack.config.ts`
```ts
import createExpoWebpackConfigAsync from "@expo/webpack-config/webpack";
import { Arguments, Environment } from "@expo/webpack-config/webpack/types";

module.exports = async function (env: Environment, argv: Arguments) {
  const config = await createExpoWebpackConfigAsync(env, argv);
  // Customize the config before returning it.
  return config;
};
```

Reply:
Thank you for your thorough response, Evan!

I did not formally benchmark the performance impact of this change, but there is no noticeable difference in compile time on my project when switching from my fork and the latest stable version of `expo-cli`.

I agree with your cost-benefit analysis of adding `ts-node` as a project dependency. And your proposed alternative solution addresses wanting to leverage TypeScript for `webpack.config.*`.

I am comfortable with closing this PR if alright with you.

Reply:
Ok, I'll add documentation on how to setup `webpack.config.ts` locally (as described above) so it's a bit easier for users to discover in the future. Thanks for raising this question and helping us get it resolved!